python/ray/experimental/channel.py

```diff
@@ -130,9 +241,11 @@ def write(self, value: Any, num_readers: Optional[int] = None):
                 before we can write again.
         """
         if num_readers is None:
-            num_readers = self._num_readers
+            num_readers = len(self._readers)
```

Comment:
I think it is a bit cleaner if we set a default `self._num_readers` in the constructor, since we expect `self._readers` or `self.is_remote()` to be constant for a given channel.

Reply:
Done.